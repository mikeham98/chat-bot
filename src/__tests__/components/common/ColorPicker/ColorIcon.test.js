import React from 'react';
import {shallow} from 'enzyme';
import ColourIcon from '../../../../app/components/common/ColorPicker/ColorIcon';

describe('ColourIcon', () => {

    let shallowedWrapper;
    let props = {};
    const wrapper = () => {
        if (!shallowedWrapper) {
            shallowedWrapper = shallow(<ColourIcon {...props}/>);
        }
        return shallowedWrapper;
    };

    beforeEach(() => {
        props = {
            color: '#1ccb9e',
            selected: false,
            onClick: jest.fn(),
        };
        shallowedWrapper = undefined;
    });

    describe('render', () => {
        it('should return a div with a className of colourIconWrapper', () => {
            expect(wrapper()).toMatchSnapshot();
        });
        it('should return a div with a className of colourIconWrapper and colourIconSelected', () => {
            props.selected = true;
            expect(wrapper()).toMatchSnapshot();
        });
    });

    describe('functionality', () => {
        describe('prop', () => {
            describe('onClick', () => {
                it('should be called on onClick of div', () => {
                    expect(props.onClick).not.toHaveBeenCalled();
                    wrapper().find('div').first().props().onClick();
                    expect(props.onClick).toHaveBeenCalledTimes(1);
                });
            });
        });
    });
});