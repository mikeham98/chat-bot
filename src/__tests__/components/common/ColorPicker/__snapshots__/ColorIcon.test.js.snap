// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColourIcon render should return a div with a className of colourIconWrapper 1`] = `
<div
  className="colourIconWrapper"
  onClick={[MockFunction]}
  style={
    Object {
      "backgroundColor": "#1ccb9e",
    }
  }
/>
`;

exports[`ColourIcon render should return a div with a className of colourIconWrapper and colourIconSelected 1`] = `
<div
  className="colourIconWrapper colourIconSelected"
  onClick={[MockFunction]}
  style={
    Object {
      "backgroundColor": "#1ccb9e",
    }
  }
/>
`;
